File: src/client/flutter/lib/models/entities/my_color.dart
import 'package:flutter/material.dart';

class MyColor {
  int id;
  String name;
  double red;
  double green;
  double blue;
  double opacity;

  MyColor({
    required this.id,
    required this.name,
    required this.red,
    required this.green,
    required this.blue,
    required this.opacity,
  });

  MyColor.fromJson(Map<String, dynamic> json)
      : id = json["id"],
        name = json["name"],
        red = double.parse(json["red"]),
        green = double.parse(json["green"]),
        blue = double.parse(json["blue"]),
        opacity = double.parse(json["opacity"]);

  Color toColor() {
    return Color.fromARGB(opacity.floor(), red.floor(), green.floor(), blue.floor());
  }

  @override
  String toString(){
    return name;
  }

}

File: src/app/android/app/src/main/java/com/cyberacy/app/models/entities/MyColor.kt
package com.cyberacy.app.models.entities

import android.graphics.Color
import com.google.gson.annotations.SerializedName

class MyColor(

    @SerializedName("id")
    val id: Int,

    @SerializedName("name")
    val name: String,

    @SerializedName("red")
    val red: Int,

    @SerializedName("green")
    val green: Int,

    @SerializedName("blue")
    val blue: Int,

    @SerializedName("opacity")
    val opacity: Int,
) {

    fun toColor(): Int {
        return Color.argb(opacity, red, green, blue)
    }

}